runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorkerTest.java

```diff
@@ -2852,6 +2853,74 @@ public void testActiveWorkForShardedKeys() throws Exception {
     Mockito.verifyNoMoreInteractions(mockExecutor);
   }
 
+  @Test
+  public void testMaxThreadMetric() throws Exception {
```

Comment:
Hi @al97 
would you mind having another look at this test? It looks flaky on Jenkins (#27555).
Thanks so much :)